### PR TITLE
Upgrade RESTEasy dependency to `6.0.0.Final` and upgrade Springframework to `6`

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

**Comment by liweinan** [comment]
This error should be fixed in Undertow:

image

**Comment by liweinan** [comment]
@jamezp During `resteasy-spring` build I get this error:

image

I can confirm the `jboss-logging` tools versions are aligned with `resteasy` main branch:

image

I have no idea how to fix this. Could you please give some advice on fixing this? Thanks!

**Comment by liweinan** [comment]
@jamezp I have built Spring `6.0.0-M2` locally:

image

And installed into my local Maven repo:

image

And did a `maven clean install` of `resteasy-spring`. However it seems the message bundle generation problem still exists:

image

It seems the `javax.*` is not from Springframework. Do you have any suggestion how could I further troubleshooting this problem?

**Comment by liweinan** [comment]
I did a `mvn dependency:tree` and I can see there are two places using `javax.*`:

image

I'll try to exclude them to see if there are any changes.

**Comment by liweinan** [comment]
@jamezp Okay I see. I need to change the `javax.*` usages in code too. I'll take action on this. Thanks for help James!

**Comment by jamezp** [comment]
Yes. You'll likely need to upgrade Arquillian as well.

**Comment by jamezp** [comment]
@liweinan Note I think you may also need to add:
```xml
<repositories>
    <!-- Temporary repository until Spring 6 is final and synchronized to Maven central -->
    <repository>
        <id>spring-repo</id>
        <name>Spring Repository</name>
        <url>https://repo.spring.io/milestone</url>
    </repository>
</repositories>
```

This can be temporary, but you won't be able to cleanly build without it.

**Comment by liweinan** [comment]
@jamezp Oh that's why I can't download the Springframework `6.0.0-M2` and built it locally. Thanks for the hint! :D

**Comment by liweinan** [comment]
Local build passed naively:

image

Two things to note:

- As James pointed out: Spring repo needs to be added to get `6.0.0-M2`
- Java 17 is needed at least by Spring 6 build

In addition, I need to resume the rest of the modules.

I'll go on working on this.

**Comment by liweinan** [comment]
@jamezp I need to switch to Java 17 for the build to pass.

**Comment by liweinan** [comment]
@jamezp The left problem is:

image

Which dependency should I replace for this?

**Comment by jamezp** [comment]
@liweinan I don't think there is a replacement for that so it needs to stick with the `javax` prefix.

**Comment by jamezp** [comment]
@liweinan see https://jakarta.ee/specifications/platform/9/jakarta-platform-spec-9.html#jakarta-management-removed

**Comment by liweinan** [comment]
@jamezp yes sir!

**Comment by liweinan** [comment]
@jamezp after upgrading Arquillian I got above build problem:

image

The error message doesn't provide much information. Do you have any suggestion how I can further troubleshooting the problem?

**Comment by jamezp** [comment]
@liweinan https://github.com/liweinan/resteasy-spring/pull/2 should fix it.

**Comment by liweinan** [comment]
@jamezp thanks for teaching!

**Comment by liweinan** [comment]
@jamezp Ready for review :D

File: testsuite/integration-tests-spring/pom.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
     </properties>
 
     <modules>
-        <module>deployment</module>
+<!--        <module>deployment</module>-->
```
Comment (jamezp) [inline testsuite/integration-tests-spring/pom.xml:23]: Does this not work? I ran it locally and the tests passed for me.

**Review by jamezp — COMMENTED** [review]
Looks good @liweinan. Just the one minor question about the `deployment` module being re-enabled.

File: testsuite/integration-tests-spring/pom.xml (hunk 1 above)
Comment (liweinan) [reply testsuite/integration-tests-spring/pom.xml:23]: oops. I missed it :p

**Comment by liweinan** [comment]
All work done. Next step I'll do the release. Thanks for help James!

**Pull request merged**